ord_schema/resolvers_test.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020 Open Reaction Database Project Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for ord_schema.units."""
+
+from absl.testing import absltest
+from absl.testing import parameterized
+from rdkit import Chem
+
+from ord_schema import resolvers
+from ord_schema.proto import reaction_pb2
+
+
+class NameResolversTest(absltest.TestCase):
+
+    def test_resolve_names(self):
+        roundtrip_smi = lambda smi: Chem.MolToSmiles(Chem.MolFromSmiles(smi))
+        message = reaction_pb2.Reaction()
+        message.inputs['test'].components.add().identifiers.add(type='NAME',
+                                                                value='aspirin')
+        self.assertTrue(resolvers.resolve_names(message))
+        resolved_smi = roundtrip_smi(
+            message.inputs['test'].components[0].identifiers[1].value)
+        self.assertEqual(resolved_smi, roundtrip_smi('CC(=O)Oc1ccccc1C(O)=O'))
+        self.assertEqual(
+            message.inputs['test'].components[0].identifiers[1].type,
+            reaction_pb2.CompoundIdentifier.IdentifierType.SMILES)
+        self.assertRegex(
+            message.inputs['test'].components[0].identifiers[1].details,
+            'NAME resolved')
+
+
+class InputResolversTest(parameterized.TestCase, absltest.TestCase):
+
+    def test_input_resolve(self):
+        string = '10 g of THF'
+        reaction_input = resolvers.resolve_input(string)
+        self.assertEqual(reaction_input.components[0].mass,
+                         reaction_pb2.Mass(value=10, units='GRAM'))
+        self.assertEqual(
+            reaction_input.components[0].identifiers[0],
+            reaction_pb2.CompoundIdentifier(type='NAME', value='THF'))
+        self.assertEqual(len(reaction_input.components), 1)
+
+        string = '100 mL of 5.0uM sodium hydroxide  in water'
```

Comment:
Hmm...this technically breaks the template since there are two spaces before "in". Do you want to allow this? If not, consider using a character class like `[a-zA-Z0-9\.]` instead of `.` in the regex.

Reply:
I don't necessarily mind extra spaces. I've removed it from the test case though.